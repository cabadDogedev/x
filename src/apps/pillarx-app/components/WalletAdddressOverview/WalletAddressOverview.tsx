import { useEffect, useState } from 'react';
import { CopyToClipboard } from 'react-copy-to-clipboard';

// images
import { MdCheck } from 'react-icons/md';

// components
import ProfileIcon from '../../images/profile-icon.svg';
import CopyIcon from '../../images/copy-icon.svg';
import Body from '../Typography/Body';

type WalletAddressOverviewProps = {
    address: string;
    className?: string;
}

const WalletAddressOverview = ({ address, className }: WalletAddressOverviewProps) => {
    const [isCopied, setIsCopied] = useState(false);

    useEffect(() => {
        if (isCopied) {
          const timer = setTimeout(() => {
            setIsCopied(false);
          }, 30000);
    
          return () => clearTimeout(timer);
        }
      }, [isCopied]);


    return (
        <div className={`flex gap-2.5 items-center ${className}`}>
            <div className='bg-[#312F3A] p-2.5 rounded-full w-10 h-10'>
                <img src={ProfileIcon} className='w-full h-full' />
            </div>
            <Body>{address.substring(0, 6)}...{address.substring(address.length - 5)}</Body>
            <CopyToClipboard text={address} onCopy={() => setIsCopied(true)}>
                <div className='w-4 h-4'>
                {isCopied ? <MdCheck style={{ width: 'full', height: 'full' }} /> : <img src={CopyIcon} className='cursor-pointer' style={{ width: 'full', height: 'full' }}/>}
                </div>
            </CopyToClipboard>
        </div>
    )
}

export default WalletAddressOverview;
