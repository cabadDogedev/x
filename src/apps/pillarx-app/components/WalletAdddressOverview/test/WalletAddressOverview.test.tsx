import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import WalletAddressOverview from '../WalletAddressOverview';

describe('<WalletAddressOverview />', () => {
  const address = '0x1234567890';
  const className = 'custom-class';
  const mockSetIsCopied = jest.fn();
  const mockUseState = jest.spyOn(React, 'useState');

  beforeEach(() => {
    mockUseState.mockReturnValue([false, mockSetIsCopied]);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly and matches snapshot', () => {
    const tree = renderer.create(<WalletAddressOverview address={address} className={className} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct icon and address', () => {
    const tree = renderer.create(<WalletAddressOverview address={address} className={className} />).toJSON() as ReactTestRendererJSON;

    const profileIcon = (tree.children?.find(child =>
      typeof child === 'object' && child.props.className === 'bg-[#312F3A] p-2.5 rounded-full w-10 h-10'
    ) as ReactTestRendererJSON);

    const addressText = (tree.children?.find(child =>
      typeof child === 'object' && child.props.children === `${address.substring(0, 6)}...${address.substring(address.length - 5)}`
    ) as ReactTestRendererJSON);

    expect(profileIcon).not.toBeNull();
    expect(addressText).not.toBeNull();
    expect(tree.props.className).toContain(className);
  });
});
