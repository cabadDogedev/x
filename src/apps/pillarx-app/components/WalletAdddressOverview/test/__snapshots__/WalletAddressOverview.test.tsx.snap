// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WalletAddressOverview /> renders correctly and matches snapshot 1`] = `
<div
  className="flex gap-2.5 items-center custom-class"
>
  <div
    className="bg-[#312F3A] p-2.5 rounded-full w-10 h-10"
  >
    <img
      className="w-full h-full"
      src={
        {
          "ReactComponent": {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "default": "profile-icon.svg",
        }
      }
    />
  </div>
  <p
    className="text-base font-medium undefined"
  >
    0x1234
    ...
    67890
  </p>
  <div
    className="w-4 h-4"
    onClick={[Function]}
  >
    <img
      className="cursor-pointer"
      src={
        {
          "ReactComponent": {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "default": "copy-icon.svg",
        }
      }
      style={
        {
          "height": "full",
          "width": "full",
        }
      }
    />
  </div>
</div>
`;
